hudi-client/hudi-spark-client/src/main/scala/org/apache/hudi/SparkFileFormatInternalRowReaderContext.scala

```diff
@@ -245,6 +241,8 @@ class SparkFileFormatInternalRowReaderContext(parquetFileReader: SparkParquetRea
         override def next(): Any = {
           (skeletonFileIterator.next(), dataFileIterator.next()) match {
             case (s: ColumnarBatch, d: ColumnarBatch) =>
+              ValidationUtils.checkState(partitionFieldAndValues.isEmpty,
+                "Columnar reads do not support bootstrap partitioning")
```

Comment:
After a second thought, this is incurred per column batch, so it will add more overhead.  If we couldn't make the validation once per instantiation of the iterator, let's remove this check here.